src/test/java/com/microsoft/sqlserver/jdbc/connection/ConnectionTest.java

```diff
@@ -42,4 +46,28 @@ public void testConnections() throws SQLException {
         try (Connection con = ds.getConnection()) {
         }
     }
+    
+    @Test
+    public void testConnectWithIPAddressPreference () throws SQLException {
+        SQLServerDataSource ds = new SQLServerDataSource();
+        ds.setURL(connectionString);
+        ds.setIPAddressPreference("IPv4First");
+        try (Connection con = ds.getConnection()) {
+        }
+        ds.setIPAddressPreference("IPv6First");
+        try (Connection con = ds.getConnection()) {
+        }
+        ds.setIPAddressPreference("UsePlatformDefault");
+        try (Connection con = ds.getConnection()) {
+        }
+        ds.setIPAddressPreference("Bogus");
+        try (Connection con = ds.getConnection()) {
+            fail(TestResource.getResource("R_expectedFailPassed"));
+        }
+        catch (Exception e){
+            MessageFormat form = new MessageFormat(TestResource.getResource("R_InvalidIPAddressPreference"));

```

Comment:
this error is not correct and totally hides the real error. Please check how other tests do this

Reply:
Added earlier check

Reply:
I've split them up and used the recommended check now.